docs/guides/securing-apps/token-exchange.adoc

```diff
@@ -6,6 +6,322 @@ title="Using token exchange"
+
+There can be multiple other use cases where token exchange can be useful, but the one above is the most typical.
+
+==== Example token exchange request
+
+The example token exchange request of the client `requester-client` in the realm `test` can look like this. Note that `subject_token` is the access token issued to the `initial-client`:
+
+[source,bash]
+----
+POST /realms/test/protocol/openid-connect/token
+Authorization: Basic cmVxdWVzdGVyLWNsaWVudDpwYXNzd29yZA==
+Content-Type: application/x-www-form-urlencoded
+Accept: application/json
+
+grant_type=urn:ietf:params:oauth:grant-type:token-exchange&
+subject_token=$SUBJECT_TOKEN&
+subject_token_type=urn:ietf:params:oauth:token-type:access_token&
+requested_token_type=urn:ietf:params:oauth:token-type:access_token
+----
+
+The example token exchange response may look like this:
+
+[source,json]
+----
+{
+  "access_token": "eyJhbGciOiJSUzI1NiIsIn...",
+  "expires_in": 300,
+  "token_type": "Bearer",
+  "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+  "session_state": "287f3c57-32b8-4c0f-8b00-8c7db231d701",
+  "scope": "default-scope1",
+  "refresh_expires_in": 0,
+  "not-before-policy": 0
+}
+----
+
+[[_standard-token-exchange-enable]]
+=== How to enable token exchange
+
+To enable the standard token exchange, it is needed to have feature `token-exchange-standard:v2` enabled, which is by default. Besides that, it is needed to enable `Standard token exchange` switch for
+the client, which is supposed to send token exchange requests. Like `requester-client` from <<_standard-token-exchange-flow,the previous example>>. Note that `requester-client` must be confidential client
+and similarly like for the other grant requests, the token exchange requests must be authenticated by the appropriate link:{adminguide_link}#_client-credentials[client authentication method] configured
+for the client.
+
+.Enabling Token Exchange
+image::token-exchange-switch.png[Enabling Token Exchange]
+
+[[_standard-token-exchange-request]]
+=== Request and response parameters
+
+The parameters are aligned with the https://datatracker.ietf.org/doc/html/rfc8693#name-token-exchange-request-and-[Token exchange specification]. Here some {project_name} specific details.
+
+grant_type::
+    _REQUIRED._ The value of the parameter must be `urn:ietf:params:oauth:grant-type:token-exchange`.
+
+subject_token::
+    _REQUIRED._ A security token that represents the identity of the party on behalf of whom the request is being made.
+
+subject_token_type::
+    _REQUIRED._ This parameter is the type of the token passed in the `subject_token` parameter. This must be `urn:ietf:params:oauth:token-type:access_token` when standard token exchange is being used
+    as {project_name} does not support any other types for the standard token exchange.
```

Comment:
```suggestion
    _REQUIRED._ This parameter is the type of the token passed in the `subject_token` parameter. This must be `urn:ietf:params:oauth:token-type:access_token` when the standard token exchange is being used
    because {project_name} does not support other types for the standard token exchange.
```